tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/Function.java

```diff
@@ -0,0 +1,194 @@
+/*
+  Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================
+ */
+package org.tensorflow.op.core;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.tensorflow.ConcreteFunction;
+import org.tensorflow.GraphFunction;
+import org.tensorflow.NamedGraphFunction;
+import org.tensorflow.Operand;
+import org.tensorflow.Operation;
+import org.tensorflow.OperationBuilder;
+import org.tensorflow.op.Scope;
+import org.tensorflow.op.annotation.Endpoint;
+import org.tensorflow.op.annotation.Operator;
+
+/**
+ * Ops for calling {@link GraphFunction}.  Even though the C API docs say the name of the Op needs to be the name of the
+ * function, they mean the type.
+ */
+@Operator
+public class Function {
+
+  /**
+   * Call {@code function}, adding it to the execution environment if it isn't already present.
+   *
+   * @param function the function to call
+   * @param inputs the inputs to the function
+   * @return the outputs of the function
+   */
+  @Endpoint(name = "callFunction")
+  public static List<Operand<?>> callFunction(Scope scope, GraphFunction function, List<Operand<?>> inputs) {
```

Comment:
Why not simply call these endpoints `call`? I like the simplicity of `tf.call(function, x).get("y")`

The function instance type is enough to differentiate the different signatures

Reply:
I had issues with `call(NamedGraphFunction, List<>)`, in that I'd like to have it return a Map using the named outputs even if named inputs aren't used, but that collides with `call(GraphFunction, List<>)` which returns a List.  Let's delay this till after the merging w/ ConcreteFunction conversation?